### PR TITLE
feat: new assertion syntax 

### PR DESCRIPTION
**Pull request by roman-sainchuk**
## What/Why/How?
This is rough implementation of new assertion syntax according to ~[docs](https://github.com/Redocly/redocly-cli/pull/857)~ [new docs](https://github.com/Redocly/redocly-cli/pull/917).

## Reference

Also closes https://github.com/Redocly/redocly-cli/issues/915

## Check yourself

- [x] Code is linted
- [x] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Review by RomanHotsiy — COMMENTED** [review]
Thanks, @roman-sainchuk! This looks good overall.

It is missing more tests but I can pick it up to add more tests later this week.

File: __tests__/lint-config/invlid-lint-config-saverity/.redocly.yaml [hunk 1]
```diff
@@ -4,8 +4,9 @@ apis:
 
 rules:
   assert/path-item-mutually-required:
-    context:
-      - type: PathItem
-    subject: Operation
+    subject:
```
Comment (RomanHotsiy) [inline __tests__/lint-config/invlid-lint-config-saverity/.redocly.yaml:7]: I just noticed that the filename has typos.

File: __tests__/lint/assertions-casing-camel-case-error/.redocly.yaml [hunk 2]
```diff
@@ -4,20 +4,34 @@ apis:
...
-    property: operationId
+    where:
+      - subject: 
+          type: Operation
+          filterInParentKeys: [get]
+          property: responses
+        assertions:
+          defined: true
+    subject: 
+      type: Operation
```
Comment (RomanHotsiy) [inline __tests__/lint/assertions-casing-camel-case-error/.redocly.yaml:15]: Can we use `filterInParentKeys` in this Operation directly without adding `where`?

File: __tests__/lint/assertions-casing-kebab-case-error/.redocly.yaml [hunk 3]
```diff
@@ -4,16 +4,26 @@ apis:
...
+    where:
+      - subject: 
+          type: Operation
+          filterInParentKeys: [get]
+          property: responses
+        assertions:
+          defined: true
+    subject: 
+      type: Operation
+      property: operationId
```
Comment (RomanHotsiy) [inline __tests__/lint/assertions-casing-kebab-case-error/.redocly.yaml:16]: same here, can we use `filterInParentKeys` directly here?

**Review by RomanHotsiy — COMMENTED** [review]
Left a few comments.

File: __tests__/lint/assertions-casing-camel-case-error/.redocly.yaml (hunk 2 above)
Comment (SmoliyY) [reply __tests__/lint/assertions-casing-camel-case-error/.redocly.yaml:15]: We can go with it. This is something that was overwritten during test refactoring

File: __tests__/lint/assertions-casing-camel-case-error/.redocly.yaml [hunk 4]
```diff
@@ -4,20 +4,34 @@ apis:
...
-    context:
-      - type: Operation
-        matchParentKeys: [get]
-    subject: Operation
-    property: operationId
+    where:
+      - subject: 
+          type: Operation
+          filterInParentKeys: [get]
+          property: responses
```
[the remark this answers is not in the document]
Comment (SmoliyY) [reply __tests__/lint/assertions-casing-camel-case-error/.redocly.yaml:11]: Yes, this make but also we should update docs with examples.

File: __tests__/lint/assertions-casing-kebab-case-error/.redocly.yaml (hunk 3 above)
Comment (tatomyr) [reply __tests__/lint/assertions-casing-kebab-case-error/.redocly.yaml:9]: Resolved

**Review by RomanHotsiy — APPROVED** [review]
LGTM 👍 🎉

**Pull request merged**